# normal distribution

In [1]:
import scipy.stats as ss

x is a normally distributed variable with mean μ =30 and standard deviation σ=4
find
a)p(x<40)
b)p(x>21)
c)p(30<x<35)


In [2]:
nd=ss.norm(100,12).cdf(98)
#prob of getting 98 in normal distribution
nd

0.43381616738909634

In [3]:
#a)
ss.norm(30,4).cdf(40)

0.9937903346742238

In [4]:
#b)
x=ss.norm(30,4).cdf(21)
1-x

0.9877755273449553

In [5]:
x=(ss.norm(30,4).cdf(35))-(ss.norm(30,4).cdf(30))
x


0.39435022633314465

2)let x be therandom varible that represents the speed of cars x has μ=90 and σ=10.we have to find the probability that x is higher than 100 or p(x>100)

In [6]:
1-ss.norm(90,10).cdf(100)


0.15865525393145707

3)let x be the random variable that represents the length of time.it has a mean of 50 and a standard deviation of 15.wehave to find the probability that x is between 50 and 70 

In [7]:
ss.norm(50,15).cdf(70)-ss.norm(50,15).cdf(50)

0.4087887802741321

4)the length of similar components produced bya company are approximated by a normal distribution model with a mean of 5cm and a standard deviation of 0.02cm.if a component is choosen at random
a)what is the probability that the length of this components is between 4.98 and 5.02cm
b)what is the probability that the length of this components is between 4.96 and 5.04cm

In [8]:
#a)
ss.norm(5,0.02).cdf(5.02)-ss.norm(5,0.02).cdf(4.98)

0.6826894921370756

In [9]:
#b)
ss.norm(5,0.02).cdf(5.04)-ss.norm(5,0.02).cdf(4.96)

0.9544997361036418

In [10]:
ss.norm(5,0.02).cdf(5.06)-ss.norm(5,0.02).cdf(4.94)

0.9973002039367396

5) the length of life of an instrument produced by a machine has a normal distribution witha mean of 12 months and std deviation of 2 months find the probability that an instrument produced by thios machine will  last
a)less than 7months
b)between 7 and 12 months


In [11]:
x=ss.norm(12,2).cdf(7)
y=ss.norm(12,2).cdf(12)-ss.norm(12,2).cdf(7)
print(x)
print(y)

0.006209665325776132
0.49379033467422384


In [12]:
from scipy.stats import binom
binom.pmf(3,5,0.75)
#binom.pmf(x,n,p)

0.26367187499999994

1) your basketball team is playing a series of 5 games against your opponent.the winner is those who wins more games (out of 5).let assume that your team is more skilled and has 75
% chance of winning .it means there is s 25% chance of losing.what is the probability of your team get 3 wins 

In [13]:
ss.binom.pmf(3,5,0.75)

0.26367187499999994

2)a box of candies has many different colors in it.there is a 15% chance of gettying a pink candy.what is the probability that exactly 4 candies in a box are pink out of 10?

In [14]:
ss.binom.pmf(4,10,0.15)

0.04009570787988284

3)lets say that 80% of all business startups in the it industry report that they generate a profit in their first year.if a sample of 10 new it business startup are selcted,find their probability that exactly seven will generate the profit?

In [15]:
binom.pmf(7,10,0.8)

0.20132659200000014

# poision distribution

1)the average number of homes sold by the acme realty company is 2 homes per day.what is the probability that exactly 3 homes will be sold tommorow?

In [16]:
ss.poisson.pmf(3,2)
#ss.poisson.pmf(x,mean)

0.18044704431548356

In [17]:
#alternate method
from scipy.stats import poisson
poisson.pmf(3,2)

0.18044704431548356

2)suppose the average number of lions seen on a 1-day safari is 5.what is the probability that tourists will see fewer than four lions on the next 1-day safari.calculate p(x<4)  

In [18]:
poisson.pmf(0,5)+poisson.pmf(1,5)+poisson.pmf(2,5)+poisson.pmf(3,5)

0.26502591529736164

#excercise:

1)wool fibre breaking strengths are normally distributed with mean (m)=23.56 newtons and std deviation=4.55.what proportion of fibers would have breaking strength of 14.45 or less?

In [19]:
ss.norm(23.56,4.55).cdf(14.45)

0.022631730991261142

In [20]:
#a)
ss.norm(195,25).cdf(180)

0.2742531177500736

In [21]:
#b)


In [22]:
#c)
ss.norm(195,25).cdf(240)-ss.norm(195,25).cdf(180)

0.6898165631370006

In [23]:
ss.binom.pmf(3,20,0.15)

0.24282889614926725

In [24]:
poisson.pmf(3,5)

0.1403738958142805

In [25]:
#a)
poisson.pmf(3,4)

0.19536681481316454

In [26]:
#b)
1-poisson.pmf(0,4)

0.9816843611112658

In [27]:
#c)
poisson.pmf(6,8)

0.12213821545677205

In [28]:
#a)
ss.norm(20,2).cdf(19.5)

0.4012936743170763

In [29]:
#b)
ss.norm(20,2).cdf(22)-ss.norm(20,2).cdf(20)

0.3413447460685429

In [30]:
#a)
1-ss.norm(70,10).cdf(80)

0.15865525393145707

In [31]:
#b)
1-ss.norm(70,10).cdf(60)

0.8413447460685429

exercise:

1)Two teams A and B are playing a series of 5 games. The team who wins the three or more games will win the series. Assume that the outcome of the games are independent and probability of winning for each team is 0.5. What is the probability that A will win the series? 


In [32]:
ss.binom.pmf(3,5,0.5)+ss.binom.pmf(4,5,0.5)+ss.binom.pmf(5,5,0.5)

0.49999999999999994

In [57]:
#alternate method
1-ss.binom.cdf(2,5,0.5)

0.5

2)In a call center 180 calls are received at random over a period of 12 hours. Then the probability to receive calls between 180 to 200 every 12 hours is: 

In [45]:
i=180
x=0
while i<201:
    x=x+poisson.pmf(i,180)
    i=i+1
print(x)



0.44470580831133655


In [42]:
#alternate method
poisson.cdf(200,180)-poisson.cdf(180,180)

0.4149841695649299

 3)A call center receives 20 call per hour on an average. Should the center be employing more people if they expect more than 25 calls in an hour, when the center cannot take more than 8 percent of the risk? 


In [47]:
1-poisson.cdf(25,20)

0.1121849727179699

4) A sample contains 1000 pieces and chances that a piece is defective in the sample is 0.2. The probability of seeing more than 200 defective piece is 


In [49]:
1-ss.binom.cdf(200,1000,0.2)

0.48108856444185344

In [55]:
#5)When rolling a die 100 times, what is the probability of rolling a 6 exactly 20 times
ss.binom.pmf(20,100,1/6)

0.06786199515844335

6)On an average the number of hits on a certain web page are 20 hits per day. What is the probability of observing more than 30 hits in a day? 


In [56]:
1-poisson.cdf(30,20)

0.01347468127992224

# T-distribution

1)Acme Corporation manufactures light bulbs. The CEO claims that an average Acme light bulb lasts 300 days. A researcher randomly selects 15 bulbs for testing. The sampled bulbs last an average of 290 days, with a standard deviation of 50 days. If the CEO's claim were true, what is the probability that 15 randomly selected bulbs would have an average life of no more than 290 days?

In [59]:
#t-distribution
#H0:null hypothesis:average life of bulbs <= 290 days
#Ha:alternate hypothesis:average life of bulbs >290 days
import numpy as np
import scipy.stats as ss
mean=300
N=15
s=50
X=290
#df=N-1=14
teststatistic=(X-mean)/(s/np.sqrt(N))
teststatistic

-0.7745966692414834

In [60]:
p=ss.t.cdf(teststatistic,14)
p

0.22573129225779137

Hence,if the true bulb life were 300 days,there is a 22.6% chance that the average bulb life for 15 randomly selcted bulbs would be greater than 290 days

2)An alternator manufacturer must produce its alternators so that they are 95% confident that it runs at less than 71.1o C under stress test in order to meet the production requirements for sale to the US government. The stress test is performed on random samples drawn from the production line on a daily basis. Today’s sample of 7 alternators has a mean of 71.3o C and a standard deviation of 0.214o . Is there a production quality issue?

In [96]:
#t-distribution
#H0:null hypothesis:average life of bulbs <71.1
#Ha:alternate hypothesis:average life of bulbs >=71.1
#one tailed test
mean=71.1
N=7
s=0.214
X=71.3
#df=N-1=6
teststatistic=(X-mean)/(s/np.sqrt(N))
teststatistic

2.4726647766959142

In [98]:
p=ss.t.cdf(teststatistic,6)
1-p

0.024141478825914442

3.A coffee shop relocates to Italy and wants to make sure that all lattes are consistent. They believe that each latte has an average of 4 Oz of espresso. If this is not the case, they must increase or decrease the amount. A random sample of 25 lattes shows a mean of 4.6 Oz of espresso and a standard deviation of .22 Oz. Use alpha = .05

In [65]:
#two tailed test
#H0:latte=4
#Ha:latte!=4
mean=4
N=25
s=0.22
X=4.6
#df=25-1=24
teststatistic=(X-mean)/(s/np.sqrt(N))
teststatistic


13.63636363636363

In [100]:
ss.t.cdf(teststatistic,24)

0.9895577662014252

In [68]:
ss.t.sf(teststatistic,24)*2 #two tail test. two sided p value

8.48542925590331e-13

In [ ]:
#reject the null hypothesis

4)A sports company claim that the average life of their product is 1500 hours.  However a sample of size 10 is tested and life is reported as 1800 1200 1300 1500 1700 1500 1200 1500 1400 1300. Can we doubt company’s claim  at 10% significance level

#H0:the average life of product=1500hrs
#Hathe average life of product  is not equal to 1500hrs


In [73]:
popmean=1500
N=10
s=201.1
sampmean=1440
#df=9       (N-1)
teststatistic=((sampmean)-(popmean))/(s/np.sqrt(N))
teststatistic

-0.9434940806071744

In [71]:
ss.t.cdf(teststatistic,9)

0.18503242571862266

In [82]:
ss.t.sf(teststatistic,9)*2   #p value (two tail test - one sample)

1.6672490349237888e-06

In [ ]:
#reject null hypothesis

5)The average mortgage rates in a city A is 8.43 percent. However it is seen in the random sample of 11 banks that the mean rate is 8.61 with a standard deviation of 0.33 percent. At 2 percent significance level can we conclude that mortgage rates are more than 8.43%?

#H0:the average mortage rate is 8.43%

#Hathe average mortage rate is more than 8.43%

In [77]:
mean=8.43
N=11
s=0.33
X=8.61
#df=25-1=24
teststatistic=(X-mean)/(s/np.sqrt(N))
teststatistic

1.8090680674665787

In [101]:
pvalue=1-ss.t.cdf(teststatistic,10)
pvalue
#significance level is 0.02
#p value is greater than significance level
# accept null hypothesis.hence the average mortage rate is 8.43%

0.016477081159621876

6)If the population the average IQ is 100. A team of scientists wants to test a new medication to see if it has either a positive of negative effect on intelligence, or no effect at all. A sample of 30 participants who have taken the medication has a mean of 140 with a standard deviation of 20. Did the medication affect intelligence?


In [80]:
mean=100
N=30
s=20
X=140
#df=30-1=29
teststatistic=(X-mean)/(s/np.sqrt(N))
teststatistic

10.954451150103322

In [81]:
ss.t.cdf(teststatistic,29)

0.9999999999959894

In [84]:
#two tailed test
ss.t.sf(teststatistic,29)*2
# p value is less than the significance level
# reject null hypothesis.medication affect intelligence

8.02128714022661e-12

7)At a traffic signal there are 3 cars arriving in a min. A random sample of 1500mins is taken and 7500 cars were observed arriving at the signal. Has the num of cars arriving at the signal increased at 5% significance level?

In [102]:
#H0:the car arriavl rate is 3 per minute
#Ha:the car arriavl rate is more than 3 per minute
#poisson distribution
popmean=3
#samplemean=7500/1500=5
p=1-ss.poisson.cdf(7500/1500,3)
p

0.08391794203130343

In [103]:
#significnce level=0.05
#accept null hypothesis

8) the mean and varience of a binomial variate are 12 and 8 respectively.choose n and p from the following
a)63,1/3
b)36,1/2
c)36,1/3
d)64,1/3


In [ ]:
#Solution:
#np=12;npq=8;q=1-p
a)63*1/3=21
b)36*1/2=18
c)36*1/3=12
npq=np(1-q)
8=36*(1/3)*(2/3)=8

ans:c